### ISSUE TITLE
Give your bug Adding a Transition = hangs the program

### ISSUE THREAD

**Issue by nanohtpc** [OpenShot/openshot-qt#3401]
when adding a video to the timeline, choosing a RANDOM transition... hangs the program...

Win x64

**Comment by SuslikV** [comment]
Not enough info to track the issue. I cannot reproduce this. Can you share OpenShot's log-file (from the session when trouble is happened)?

**Comment by nanohtpc** [comment]
OpenShot-v2.5.1-dev2-1586151841-166f6938-232f3f79-x86_64

Was the last version that doesnt have problems, but beyond that, if i tried to add a random transition OR try to load a OSP that have a ramdom one, it just hangs/crash the program....

![Screenshot (243)](https://user-images.githubusercontent.com/34278444/80128058-c6f95000-855a-11ea-8bb7-44b9564a01ad.png)

win x64

it happens with all kind of videos...

**Comment by nanohtpc** [comment]
that was a perfect OSP with older version, but hangs with the last 8 new ones...

also happens trying to add new files... only with random transitions....

**Comment by SuslikV** [comment]
Hmm... It's far deeper issue.

**Comment by nanohtpc** [comment]
OpenShot-v2.5.1-dev2-1586151841-166f6938-232f3f79-x86_64

was the last version that works...

**Comment by SuslikV** [comment]
Issue caused by next line of code:
https://github.com/OpenShot/openshot-qt/blob/38be10411756cbc76447afb0a1d1b202ededddde/src/classes/app.py#L110

Try to remove it like:
```
        self.setApplicationVersion(info.SETUP['version'])
        self.setWindowIcon(QIcon(":/openshot.svg"))
        try:
```
after
```
        self.setApplicationVersion(info.SETUP['version'])
        try:
```
Use _notepad++_ application (minimalist version is enough for your needs). Save file, restart application, try again.

-------------

PR in question: https://github.com/OpenShot/openshot-qt/pull/3354

**Comment by SuslikV** [comment]
Usually, when error occurs, openshot log entry has next info:
```
...
add_to_timeline:INFO accept
  exceptions:ERROR Unhandled Exception
Traceback (most recent call last):
  File "I:\Temp\OpenShot-dev\01\{app}\windows\add_to_timeline.py", line 328, in accept
    transition_reader = openshot.QtImageReader(transition_path)
  File "C:\Users\Administrator\builds\7546b651\0\OpenShot\openshot-qt\openshot.py", line 2501, in __init__
RuntimeError: File could not be opened.
...
```

**Comment by SuslikV** [comment]
@ferdnyc can you say what is special with the Windows daily builds of openshot? If I place this string `self.setWindowIcon(QIcon(":/openshot.svg"))` after the:

https://github.com/OpenShot/openshot-qt/blob/38be10411756cbc76447afb0a1d1b202ededddde/src/classes/app.py#L229-L230

all works OK for daily builds under Windows. My own build/freeze, has no difference in this part at all.

**Comment by ferdnyc** [comment]
@SuslikV 

Not sure, I can try to investigate. But I'm working an offline job all day, so I'm not around.

I'm on my phone so it's hard to get the full picture of this bug, maybe I missed something. But I don't understand what the application icon has to do with a crash in the transition view. That's what this is about, right?

**Comment by SuslikV** [comment]
> But I don't understand what the application icon has to do with a crash in the transition view. That's what this is about, right?

Revert the change and ask user to confirm if new daily builds work OK.

I don't see how it works at all (in linux) if icon is set differently for different windows.

**Comment by nanohtpc** [comment]
OpenShot-v2.5.1-dev2-1589067618-6e7ad231-a62505ca-x86_64.exe (64-bit)

Problem persist...

**Comment by SuslikV** [comment]
@did you tried the solution posted above (remove the string)?

**Comment by nanohtpc** [comment]
didnt see it, cant find the file app.py...

im on windows BTW

**Comment by SuslikV** [comment]
`classes/app.py`

**Comment by nanohtpc** [comment]
what file should i modify, and where is it?

**Comment by SuslikV** [comment]
how you are looking for files?

**Comment by nanohtpc** [comment]
on this paths...

C:\Users\admin\.openshot_qt

and

D:\- Software -\OpenShot Video Editor

**Comment by SuslikV** [comment]
By second path. And you can use search abilities of the OS...

**Comment by nanohtpc** [comment]
which is the file that im looking for?

**Comment by SuslikV** [comment]
You are not looking for anything. But if you wish to make official daily builds to work normal for you, you can find the `app.py` file and follow the instructions I gave to you in https://github.com/OpenShot/openshot-qt/issues/3401#issuecomment-626137795 (open the link in browser to see the details - this the comment to the thread you have opened on github).

**Comment by nanohtpc** [comment]
easy man... im looking. but doesnt show app.py,

whats wrong with you to answers like that? the emoticon watergate?

i saw taht ur not in the best of mood lately, that why a lot of users
complain, u dont have a method to solve the problems, me, im a programmer
for 40 years, im cns, i i worked on the congress of my country, im not
complaining, just asking, where is teh file, or which name,

you are starting tto think that every body know the software like you do,
its the same if i think that you could understand one of my system in
almost 4 decades,, that why im asking...

relay y dont like your answered, what are you gonna tellme now? go find
another software?

if ur a e a programmer , you should have patience, you lose it, the bug
will come like a picnic.... experience here... maybe you really need a
break, or delegate the support to some body for a period of time...

u dont know me, im older than you easily, the first frase in your answer,
sounds like a i think that the word its tantrun...

chilllllllllllllllll your alive, other not.....

**Comment by SuslikV** [comment]
> whats wrong with you to answers like that? the emoticon watergate?..

My own life. "Try walking in my shoes". Sorry if it hurts you. It wasn't intentional. Language barrier also...

And you reported same issue twice. There are no lazy programmers, but people have there own lives. Developers are aware of the bug.

Daily build are poorly tested, always. My own builds, for example, has no issues with the "code" itself. So, the change is legit. If something not works, you know... there is always place for improvement. You can use release version of the application if current "daily" bothers you a lot. Last was v2.5.1.

Some additional info (that is related to the question about the file) is on the screenshot:

![OpenShot file-path screenshot brief explanation](https://user-images.githubusercontent.com/19683044/81505956-c0f5c400-92fb-11ea-98dc-a9db058cb8de.png)

I don't have better explanation.


> what are you gonna tellme now? go find another software?

And sure, what sense to "stuck" at not working software?.. Fetish?

**Comment by SuslikV** [comment]
The .exe file of the official Windows build is modified by third party util (not OS tool). This may has its consequences:
https://github.com/OpenShot/openshot-qt/blob/2d0108f8189c3ea0d725b92dafa1d351db1981a8/installer/build-server.py#L485

I have no other ideas right now. Exact version of the tool and same environment is needed to test all the things. I only know that .exe is modified not only in version string. Testers are welcomed. Other solution was known, look here: https://github.com/OpenShot/openshot-qt/pull/2772 (if this the case).

**Comment by nanohtpc** [comment]
so you´re saying that the open shot installer is modified in windows and
that makes the random transition doesnt work???????

**Comment by SuslikV** [comment]
I'm placed custom transition file in the application's folder and changed "random" path to this fixed file (no actual path) and daily build fail to load it until I removed mentioned string from the `app.py` file. This is only the file that causes the issue on my machine for the mentioned daily build. For my own build, I see no difference at all. And I cannot reproduce issue with my build (Windows).

**Comment by nanohtpc** [comment]
i got an idea...

what if you use a different version of windows to build or compile the
latest daily???

you can do it in a VM for example...

**Comment by SuslikV** [comment]
Hmm... If question was to me - I'm not a developer. I'm using _appveyor_ service to get my test builds.

**Comment by nanohtpc** [comment]
i found something new....

add_to_timeline:INFO accept
  exceptions:ERROR Unhandled Exception
Traceback (most recent call last):
  File "D:\- Software -\OpenShot Video Editor\windows\add_to_timeline.py",
line 328, in accept
    transition_reader = openshot.QtImageReader(transition_path)
  File
"C:\Users\Administrator\builds\7546b651\0\OpenShot\openshot-qt\openshot.py",
line 2501, in __init__
RuntimeError: File could not be opened.

*-
shows everytime that i want o ADD a random transition...
*-

but when i tried to LOAD a project WITH random transitions ...

main_window:INFO canceled
 main_window:INFO actionRemove_from_Project_trigger
project_data:INFO _set key: ['files', {'id': '1C4NUVDEE1'}] values: None
add: False partial: False remove: True
timeline_webview:INFO SortItems
timeline_webview:INFO UpdateLayerIndex
 files_model:INFO updating files model.
 files_model:INFO updating files model.
 main_window:INFO updateStatusChanged
preview_thread:INFO refreshFrame
preview_thread:INFO self.player.Position(): 1
 main_window:INFO Clear all thumbnails:
C:\Users\admin\.openshot_qt\thumbnail
 main_window:INFO Clear all animations: C:\Users\admin\.openshot_qt\blender
 main_window:INFO Clear all titles: C:\Users\admin\.openshot_qt\title
project_data:INFO Setting default profile to HD 720p 29.97 fps
project_data:INFO Loading project file: D:/- Streaming -/- Openshot -
Projects -/00_INTRO.osp
      assets:INFO Using existing asset folder D:/- Streaming -/- Openshot -
Projects -\00_INTRO_assets
      assets:INFO Using existing asset folder D:/- Streaming -/- Openshot -
Projects -\00_INTRO_assets
      assets:INFO Using existing asset folder D:/- Streaming -/- Openshot -
Projects -\00_INTRO_assets
project_data:INFO checking project files...
project_data:INFO checking file D:\- Streaming -\Streaming - Setup\_ MP3
_\Bensound-Acousticbreeze.mp3
project_data:INFO checking file D:\- Streaming -\Streaming - Setup\_ Videos
_\00_INTRO_\04_STREAMING_InMemorian.mp4
project_data:INFO checking file D:\- Streaming -\Streaming - Setup\_ Videos
_\00_INTRO_\03_STREAMING_Location.mp4
project_data:INFO checking file D:\- Streaming -\Streaming - Setup\_ Videos
_\00_INTRO_\02_STREAMING_Event.mp4
project_data:INFO checking file D:\- Streaming -\Streaming - Setup\_ Videos
_\00_INTRO_\01_STREAMING_Intro.mp4
project_data:INFO Project data: openshot 2.4.4-dev2, libopenshot 0.2.3-dev1

**Comment by SuslikV** [comment]
Bad news. New daily (for example: OpenShot-v2.5.1-dev2-1590303623-972c2905-7cec1822-x86_64 for Windows) are broken even more. No Preview. Just black screen...

**Comment by nanohtpc** [comment]
OMG!!!!

before i lost all my projects...

its there a way to replace all the RANDOM transitions. by editing the .OSP
in like notepad++?

something like

search RANDOM, Replace to Fade In, for example...

**Comment by SuslikV** [comment]
@nanohtpc just use release version v2.5.1.

**Comment by jonoomph** [comment]
This is fixed in the latest daily build: https://www.openshot.org/download#daily

**Comment by nanohtpc** [comment]
thanks FINALLY!!!..

btw , did u check the error in the help section? it there at least 6
months...

is the format of the last entry

help - about - changelog...
look at the top. the last character in the date, is moved!! 8)))

**Comment by ferdnyc** [comment]
@nanohtpc 

Not sure I'm seeing what you're referring to, is the issue showing in this screenshot?

![image](https://user-images.githubusercontent.com/538020/86917250-c18bba80-c0f2-11ea-9373-73edeb84c066.png)

**Comment by nanohtpc** [comment]
![image](https://user-images.githubusercontent.com/34278444/86939431-2d781e00-c107-11ea-86ca-dda0216b14c3.png)

**Comment by ferdnyc** [comment]
@nanohtpc 

Heh! Thanks.

It's a safe bet that only happens under Windows, as I'm sure it has something to do with the fact that the changelog data files are encoded UTF-16 (double-wide characters) under Windows (only).

I'll take a quick peek to see if I can figure out what's going on there. If it's something obvious and quick then great, but I probably won't look too hard.

**Comment by ferdnyc** [comment]
It was exactly what I expected, and turned out it was an easy fix, which is in #3601. Thanks for letting us know!
